### PR TITLE
obey knime limits for cpu + memory

### PR DESCRIPTION
**Pull request by maierl**
(cherry picked from commit 627fa1eeb4666af39d11490f624872c9ebc744d7)

### CODE REVIEW

**Comment by maierl** [comment]
sorry for the late bugfixes, I'll check that more thoroughly next time

File: org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java [hunk 1]
```diff
@@ -381,6 +381,11 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
         // build process with project and images
         ProcessBuilder pB = new ProcessBuilder(inFiles);
 
+        // limit cpu + memory usage
+        final Map<String, String> env = pB.environment();
+        env.put("LAZYFLOW_THREADS", String.valueOf(KNIMEConstants.GLOBAL_THREAD_POOL.getMaxThreads()));
```
Comment (dietzc) [inline org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java]: shouldn't this be a property of the node?  I don't see the relation between the KNIME threadpool and Ilastik here.

File: org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java [hunk 2]
```diff
@@ -381,6 +381,11 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
...
+        env.put("LAZYFLOW_TOTAL_RAM_MB", String.valueOf(Runtime.getRuntime().maxMemory() / 1024 / 1024));
```
Comment (dietzc) [inline org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java]: should also be a property of the node, right?

**Comment by maierl** [comment]
you mean to be able to set this manually in the config window?
that would be the next step ;)

in the current state, it's at least limited roughly so it does not run havoc on a cluster

**Comment by dietzc** [comment]
:+1:
anyway, we should add this to the dialog, right?

**Comment by maierl** [comment]
that would be better for sure

**Comment by dietzc** [comment]
great! thanks @beschu can you double check with the test-flow? Merged on master for simplicity.

**Pull request merged**